realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -95,10 +96,13 @@ public DynamicRealmObject(RealmObject obj) {
 
     /**
      * Returns the {@code boolean} value for a given field.
+     * If the field is nullable use {@link #isNull(String)} to check for {@code null} instead of using
```

Comment:
Yes, that definitely also has a lot of merit. However the only reason we went with `@Required` was because the `Nullable/NonNull/<insert other>` was overloaded by too many other libraries/frameworks. I think we all agreed that `Nullable/NotNull` was a lot better in term of expressing intent? That still leaves the question what we should use when describing what a method actually does.

Ping rest of @realm/java. What do you think?